File: BST/PopulateIOS/popu.py
class Solution:
    def prepareInVector(self, root, ans):
        if not root:
            return

        self.prepareInVector(root.left, ans)
        ans.append(root)
        self.prepareInVector(root.right, ans)

    def populateNext(self, root):
        check = []
        self.prepareInVector(root, check)

        n = len(check)

        for i in range(n - 1):
            check[i].next = check[i + 1]

        if n > 0:
            check[n - 1].next = None

File: BST/PopulateIOS/popu.cpp
#include <iostream>
#include <vector>
using namespace std;

struct Node
{
    int data;
    struct Node *left;
    struct Node *right;
    struct Node *next;

    Node(int x)
    {
        data = x;
        left = NULL;
        right = NULL;
        next = NULL;
    }
};

class Solution {
  public:
    void prepareInVector(Node* root,vector<Node*>& ans){
        if(!root)return;
        prepareInVector(root->left,ans);
        ans.push_back(root);
        prepareInVector(root->right,ans);
    }

    void populateNext(Node *root) {
        // code here
        vector<Node*>check;
        prepareInVector(root,check);
        int n = check.size();
        Node* pred = NULL;
        Node* succ = NULL;
        for(int i=0;i<n-1;i++){
            check[i]->next = check[i+1];
        }
        check[n-1]->next = NULL;
        return;
    }
};